dotnet/src/Planners/Planners.Handlebars/Handlebars/HandlebarsPlan.cs

```diff
@@ -46,11 +54,28 @@ public override string ToString()
     /// <param name="arguments">The arguments.</param>
     /// <param name="cancellationToken">The cancellation token.</param>
     /// <returns>The plan result.</returns>
-    public string Invoke(
+    public async Task<string> InvokeAsync(
         Kernel kernel,
-        KernelArguments arguments,
+        KernelArguments? arguments = null,
         CancellationToken cancellationToken = default)
     {
-        return HandlebarsTemplateEngineExtensions.Render(kernel, this._template, arguments, cancellationToken);
+        var templateFactory = new HandlebarsPromptTemplateFactory(options: HandlebarsPlanner.PromptTemplateOptions);
+        var promptTemplateConfig = new PromptTemplateConfig()
+        {
+            Template = this._template,
+            TemplateFormat = HandlebarsPromptTemplateFactory.HandlebarsTemplateFormat,
+            Name = "ExecutePlan",
+        };
+
+        var handlebarsTemplate = templateFactory.Create(promptTemplateConfig) as HandlebarsPromptTemplate;
+        try
+        {
+            return await handlebarsTemplate!.RenderAsync(kernel, arguments, cancellationToken).ConfigureAwait(false);
+        }
+        catch (HandlebarsRuntimeException ex) when (ex.Message.Contains(HallucinatedHelpersErrorMessage))
```

Comment:
This is the only way to check for what we need? What happens if handlebars fixes a typo or otherwise tweaks this message?

Reply:
<img width="757" alt="Screenshot 2023-12-07 at 8 37 25 AM" src="https://github.com/microsoft/semantic-kernel/assets/125500434/86ea5514-1ed7-4ac3-85d2-6097b0345a07">

Doesn't look like there's an error code or something else that we can check.

This is more or less an optimization to make the error scenario clearer to the user (`Helper that cannot be resolved` vs explicitly calling out `Hallucinated helpers`), so I can remove this conditional, and it would more or less have the same outcome if that's preferred.